### PR TITLE
Properly implement DM room creation & retrieval

### PR DESCRIPTION
**Pull request by fvjosef21**
This is a bug fix for the send_message, get_dm_rooms, and create_dm_room functions.

I also added code to do a lazy check if there are rooms that are on the server, but not on the bot. I am not sure
why this is happening.

I added a call to set the m.direct type on the newly created dm_room. This is not really supported by
the nio-sdk library, which is missing the MDirectResponse type.

### CODE REVIEW

File: src/niobot/client.py [hunk 1]
```diff
@@ -813,11 +831,24 @@ async def send_message(
...
+                        break
+                    else:
+                        logging.warning(f"room {r_id} not found in bot.rooms")
+            if not room:
+                logging.info("creating dm room for user {_user}")
+                response = await self.create_dm_room(_user)
+
+                room = self.rooms.get(response.room_id)
+                if not room:
+                    raise RuntimeError("DM room %r was created, but could not be found in the room list!" % room_id)
```
Comment (nexy7574) [inline src/niobot/client.py:851]: Could you add documentation for the RuntimeError exception into the docstring? only ValueError and MessageException are documented at the moment

File: src/niobot/client.py [hunk 2]
```diff
@@ -769,9 +769,27 @@ async def create_dm_room(
...
+        dm_rooms = {}
+        dm_rooms[user_id] = [result.room_id]
+
+        logging.debug(f"create_dm_user: user_id {user_id} self.user_id {self.user_id}")
+        # Trying to send m.direct type eventfrom nio.responses import DirectRoomsErrorResponse, DirectRoomsResponse
+        logging.debug(f"create_dm_user: setting m.direct type with rooms {nio.Api.to_json(dm_rooms)}")
+        await self._send(
+            nio.DirectRoomsResponse,
+            'PUT',
+            nio.Api._build_path(    ["user", self.user_id, "account_data", "m.direct"],
```
Comment (nexy7574) [inline src/niobot/client.py:787]: The whitespace is a little odd here - could you run `ruff check --fix` and `ruff format` & push on your fork? just for styling consistency

File: src/niobot/client.py [hunk 3]
```diff
@@ -769,9 +769,27 @@ async def create_dm_room(
...
             is_direct=True,
             invite=[user_id],
+            visibility=nio.api.RoomVisibility.private,
+            preset=nio.api.RoomPreset.trusted_private_chat,
         )
         if isinstance(result, nio.RoomCreateError):
             raise GenericMatrixError("Failed to create DM room", response=result)
+        
+        dm_rooms = {}
+        dm_rooms[user_id] = [result.room_id]
```
Comment (nexy7574) [inline src/niobot/client.py:779]: Just being pedantic here but wouldn't it be cleaner to just do `dm_rooms = {user_id: [result.room_id]}`?

**Review by nexy7574 — CHANGES_REQUESTED** [review]
Looks good, just a couple (mainly formatting) things that need doing before this can be merged :)

**Comment by fvjosef21** [comment]
Fixed the comments. Not sure how to submit PR again.

**Comment by nexy7574** [comment]
No need to re-submit, changes are automatically added to the PR :)

**Review by nexy7574 — APPROVED** [review]
Looks good!

**Comment by nexy7574** [comment]
All checks passed, merging.

**Pull request merged**